kore/src/Kore/Step/Axiom/Evaluate.hs

```diff
@@ -60,11 +61,13 @@ import Kore.Variables.Fresh
 evaluateAxioms
     :: forall variable simplifier
     .  (SimplifierVariable variable, MonadSimplify simplifier)
-    => [EqualityRule Variable]
+    => StepContext
```

Comment:
I'm skeptical about this; `evaluateAxioms` can only be used in a function or simplification context because its arguments are `EqualityRule`s.

Reply:
True, but we still need to distinguish between the two cases (Function evaluation vs. simplification).

Reply:
Yes, but why do we distinguish by using a context, as opposed to saying what we want? Why is it better to tell the lower levels "You are doing OnePath stuff right now, so do whatever's right for that" as opposed to "You should only accept function-like stuff when applying rules"?

If you do the second, then the higher-level code knows about high-level concepts like OnePath, function evaluation, All Path and similar stuff, while the lower level code knows about lower level concepts like "we can only handle function-like stuff".